accounts/account_manager.go

```diff
@@ -147,6 +147,16 @@ func (am *Manager) Sign(addr common.Address, hash []byte) (signature []byte, err
 	return crypto.Sign(hash, unlockedKey.PrivateKey)
 }
 
+// VerifyKeyAndSign signs hash if the private key matching the given address
+// can be decrypted with the given authKey.
+func (am *Manager) VerifyKeyAndSign(addr common.Address, keyAuth string, hash []byte) (signature []byte, err error) {
+	_, key, err := am.getDecryptedKey(Account{Address: addr}, keyAuth)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Please add a `defer zeroKey(key.PrivateKey)` here to make sure the key doesn't remain in main memory.